core/src/main/java/org/apache/brooklyn/core/entity/AbstractApplication.java

```diff
@@ -152,15 +153,19 @@ public void start(Collection<? extends Location> locations) {
         setExpectedStateAndRecordLifecycleEvent(Lifecycle.STARTING);
         try {
             preStart(locationsToUse);
-            // if there are other items which should block service_up, they should be done in preStart
-            ServiceStateLogic.ServiceNotUpLogic.clearNotUpIndicator(this, Attributes.SERVICE_STATE_ACTUAL);
-            
+
+            // Opportunity to block startup until other dependent components are available
+            Object val = config().get(START_LATCH);
+            if (val != null) log.debug("{} finished waiting for start-latch; continuing...", this);
+
             doStart(locationsToUse);
             postStart(locationsToUse);
+
+            ServiceStateLogic.ServiceNotUpLogic.clearNotUpIndicator(this, Attributes.SERVICE_STATE_ACTUAL);
```

Comment:
i guess this is how we do it elsewhere, we basically assume if `start` failed without errors then it should normally be up, and we'll get told subsequently + periodically if that isn't the case.

in which case it makes sense to me to put this after, else we might catch a rogue not-up indicator from before the start which isn't reset in time.

(but still worth a comment in the code!)